landoapi/transplants.py

```diff
@@ -896,6 +929,21 @@ def run_landing_checks(stack_state: StackAssessmentState) -> StackAssessment:
     return assessment
 
 
+def get_parsed_diffs(
+    phab: PhabricatorClient, stack_data: RevisionData
+) -> dict[int, dict]:
+    """Return a mapping of diff PHID to raw `diff --git` content."""
+    raw_diffs = {}
+    for diff in stack_data.diffs.values():
+        diff_id = phab.expect(diff, "id")
+        raw_diff = phab.call_conduit("differential.getrawdiff", diffID=diff_id)
```

Comment:
I think it makes sense to cache this on first request. It could significantly slow things down and increase traffic to Phabricator otherwise. It should be easy enough to cache by diff ID since that should be unique. I think as a follow up (perhaps in the new lando, when we will have better access to cron jobs) we should have a cache-warming job that prepopulates the cache.

In the future, we should also cache this on the Phabricator end (maybe with nginx) since I don't think Phabricator caches the `getrawdiff` response either. That might help speed up the first request from Lando in the future.

Reply:
Makes sense to me, added caching in the next iteration.